platform/security/rest/security-rest-cxfwrapper/src/main/java/org/codice/ddf/cxf/paos/PaosInInterceptor.java

```diff
@@ -0,0 +1,399 @@
+                                    "Unable to determine IdP appropriately. ECP connection will fail. SP may be incorrectly configured. Contact the administrator for the remote system.");
+                        }
+                    } else if ("Request".equals(soapHeaderElement.getLocalName())
+                            && soapHeaderElement.getNamespaceURI()
+                            .equals("urn:liberty:paos:2003-08")) {
+                        responseConsumerURL = soapHeaderElement.getAttribute("responseConsumerURL");
+                        messageId = soapHeaderElement.getAttribute("messageID");
+                    }
+                }
+                String token = createToken(authorization);
+                checkAuthnRequest(soapMessage);
+                Element authnRequestElement = SamlProtocol.getDomElement(soapMessage.getEnvelope()
+                        .getBody()
+                        .getFirstChild());
+                if (idpEntry != null) {
+                    String loc = idpEntry.getLoc();
+                    String soapRequest = buildSoapMessage(token, relayState, authnRequestElement,
+                            null);
+                    HttpResponseWrapper httpResponse = getHttpResponse(loc, soapRequest);
+                    InputStream httpResponseContent = httpResponse.content;
+                    SOAPPart idpSoapResponse = SamlProtocol.parseSoapMessage(
+                            IOUtils.toString(httpResponseContent));
+                    Iterator responseHeaderElements = idpSoapResponse.getEnvelope()
+                            .getHeader()
+                            .examineAllHeaderElements();
+                    String newRelayState = "";
+                    while (responseHeaderElements.hasNext()) {
+                        SOAPHeaderElement soapHeaderElement = (SOAPHeaderElement) responseHeaderElements.next();
+                        if ("Response".equals(soapHeaderElement.getLocalName())) {
+                            String assertionConsumerServiceURL = soapHeaderElement.getAttribute(
+                                    "AssertionConsumerServiceURL");
+                            if (!responseConsumerURL.equals(assertionConsumerServiceURL)) {
+                                String soapFault = buildSoapFault("ecp:Response",
+                                        "The responseConsumerURL does not match the assertionConsumerServiceURL.");
+                                httpResponse = getHttpResponse(responseConsumerURL, soapFault);
+                                message.setContent(InputStream.class, httpResponse.content);
+                                return;
+                            }
+                        } else if ("RelayState".equals(soapHeaderElement.getLocalName())) {
+                            newRelayState = DOM2Writer.nodeToString(soapHeaderElement);
+                            if (StringUtils.isNotEmpty(relayState) && (
```

Comment:
You are correct. This is verbatim out of the spec and I debated this for a while. I'll simplify it and just add a log message for the case where they don't match.